startup.sh

```diff
@@ -2,40 +2,67 @@
 
 set -euo pipefail
 
-function wait_for {
-  local service="$1"
-  local url="$2"
-  local expected_code="${3:-200}"
-
-  echo -n "Waiting for $service "
-  until test "$expected_code" = $(curl --output /dev/null --silent --write-out '%{http_code}' "$url"); do
-    echo -n "."
-    sleep 1
-  done
-  echo " READY"
-}
-
+BUILD_DIR=".local_yaml"
+PKI_DIR=".local_pki"
+COMPONENTS="proxy-node-gateway proxy-node-translator stub-connector"
 PN_PROJECT_DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
-PKI_OUTPUT_DIR="${PN_PROJECT_DIR}"/.local_pki
-
-pushd "${PN_PROJECT_DIR}/pki"
-  rm -f "${PKI_OUTPUT_DIR}/*"
-  bundle install --quiet
-  bundle exec generate \
-    --hub-entity-id "https://dev-hub.local" \
-    --idp-entity-id "http://stub_idp.acme.org/stub-idp-demo/SSO/POST" \
-    --proxy-node-entity-id "http://proxy-node" \
-    --hub-response-url "http://localhost:6100/SAML2/SSO/Response/POST" \
-    --idp-sso-url "http://localhost:6200/stub-idp-demo/SAML2/SSO" \
-    --proxy-sso-url "http://localhost:6100/SAML2/SSO/POST" \
-    --env \
-    "${PKI_OUTPUT_DIR}" \
-    --softhsm
-popd
-
-docker-compose up $@ -d
-
-wait_for "Gateway"  localhost:6601/healthcheck
-wait_for "Translator" localhost:6661/healthcheck
-wait_for "Stub Connector" localhost:6667/healthcheck
+PKI_OUTPUT_DIR="${PN_PROJECT_DIR}/${PKI_DIR}"
+
+# boot a local k8s cluster
+(minikube status | grep -i running) || minikube start --memory 4096
+
+# clean
+# if [[ -e "${BUILD_DIR}" ]]; then
+# 	kubectl delete -R -f "${BUILD_DIR}" || echo 'fine, continue'
+# 	rm -rf "${BUILD_DIR}"
+# fi
+# if [[ -e "${PKI_DIR}" ]]; then
+# 	kubectl delete -R -f "${PKI_DIR}" || echo 'fine, continue'
+# 	rm -rf "${PKI_DIR}"
+# fi
+
+# build all the images locally
+# send all the images to the dockerdaemon in minikube
+mkdir -p "${BUILD_DIR}"
+for component in $COMPONENTS; do
+	tag="local-$(tar c $component | md5sum | awk '{print $1}')"
+	image="govukverify/${component}:${tag}"
+	echo "building $component as $image"
+	if (eval $(minikube docker-env) && docker inspect --type=image "${image}" >/dev/null 2>&1); then
```

Comment:
`minikube docker-env` uses the `SHELL` environment variable to generate its output. This breaks for anyone not using `bash` as their shell because it seems to inherit the `SHELL` variable from the environment. So if I'm using `fish`, `minikube docker-env` seems to produce output using fish syntax even though the script is running in `bash`. So might want to stick an `export SHELL=/bin/bash` at the top of the script just to be safe.

Reply:
Resolved by explicitly asking for the bash script from `minikube docker-env --shell=bash`